client/pages/api/taskApi.ts

```diff
@@ -0,0 +1,23 @@
+import ITask from '../../common/types/ITask';
+
+export const createNewTask = async (task: ITask) => {
+  try {
+    const newTask = await fetch(`${process.env.BASEURL}/task`, {
+      method: 'POST',
+      body: JSON.stringify(task),
+      headers: { 'Content-Type': 'application/json' },
+    });
+    return await newTask.json();
```

Comment:
this shouldn't use a then instead of await newTask? and treat the data in the response and then return new task?